src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -326,6 +325,12 @@ stop_metadata_sync_to_node(PG_FUNCTION_ARGS)
 		}
 	}
 
+	bool localOnly = false;
```

Comment:
Hmm, I think our `stop_metadata_sync_to_node` implementation is missing an important step.

Instead of doing `SetWorkerColumn` on the other nodes, we should simply send the `DELETE FROM pg_dist_node` command.

I suggest we undo changes in this function, and fix the issue in a separate PR. In the end, we should not need to call `SetWorkerColumn`  as the nodes should have already been dropped

Reply:
Sorry, this os a pretty wrong suggestion :) What we have on the PR makes sense.